invenio_rdm_records/services/pids/providers/datacite.py

```diff
@@ -178,10 +185,19 @@ def update(self, pid, record, url=None, **kwargs):
         :param record: the record metadata for the DOI.
         :returns: `True` if is updated successfully.
         """
+        hide = False
+        if isinstance(record, ChainObject):
+            if record._child["access"]["record"] == "restricted":
+                hide = True
+        elif record["access"]["record"] == "restricted":
+            hide = True
+
         try:
-            # Set metadata
-            doc = self.serializer.dump_obj(record)
-            self.client.api.update_doi(metadata=doc, doi=pid.pid_value, url=url)
+            if hide:
```

Comment:
Shouldn't be able to update a doi even if it is is hidden? Also, how do I restore a hidden doi? Currently, if I change from restricted to public then the update method will update the metadata but not make the doi findable again, unless I miss something...

Reply:
yes, good catch. it should be that the hided should be put back to "findable" after a change from "restricted" to "public". i will have a look.

Reply:
That's going to be a tricky one, since I don't think we want to run `show_doi` for every single public DOI update.

I can think of two solutions:
- I can add a `status` option to the `update_doi` function in the datacite library. Then I could pass along all the updates to DataCite in one call
- We can modify pidstore so it has a `hidden` state. Then we could check locally whether the state of the DOI needs to be changed.

Let me know what you think or if you want to chat.

Reply:
@zzacharo @tmorrell what do you think of the proposed solution to this problem? this does not make it necessary to keep track the state.

Reply:
Yup, I think that should work too. Just make sure DataCite doesn't complain about having the event as "publish" for a published doi. Looks like the tests need to updated too

Reply:
If DataCite indeed doesn't complain about the "publish" events that should be good. Otherwise, we need some kind of state, maybe the caller passes the old record value, to call the publish/show doi only on the change of permission.

Reply:
How i tested it: DataCite doesn't complain.